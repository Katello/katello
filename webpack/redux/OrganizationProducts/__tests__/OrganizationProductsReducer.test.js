import { testReducerSnapshotWithFixtures } from 'react-redux-test-utils';

import {
  ORGANIZATION_PRODUCTS_REQUEST,
  ORGANIZATION_PRODUCTS_SUCCESS,
  ORGANIZATION_PRODUCTS_FAILURE,
} from '../OrganizationProductsConstants';
import reducer from '../OrganizationProductsReducer';

const fixtures = {
  'should return the initial state': {},
  'should handle ORGANIZATION_PRODUCTS_REQUEST': {
    action: {
      type: ORGANIZATION_PRODUCTS_REQUEST,
    },
  },
  'should handle ORGANIZATION_PRODUCTS_SUCCESS': {
    action: {
      type: ORGANIZATION_PRODUCTS_SUCCESS,
      payload: {
        results: ['some', 'results'],
      },
    },
  },
  'should handle ORGANIZATION_PRODUCTS_FAILURE': {
    action: {
      type: ORGANIZATION_PRODUCTS_FAILURE,
      payload: new Error('some error'),
    },
  },
};

describe('OrganizationProducts reducer', () => testReducerSnapshotWithFixtures(reducer, fixtures));
