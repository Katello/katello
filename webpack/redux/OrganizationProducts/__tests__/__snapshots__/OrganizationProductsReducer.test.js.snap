// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizationProducts reducer should handle ORGANIZATION_PRODUCTS_FAILURE 1`] = `
Object {
  "error": [Error: some error],
  "loading": false,
  "results": Array [],
}
`;

exports[`OrganizationProducts reducer should handle ORGANIZATION_PRODUCTS_REQUEST 1`] = `
Object {
  "error": null,
  "loading": true,
  "results": Array [],
}
`;

exports[`OrganizationProducts reducer should handle ORGANIZATION_PRODUCTS_SUCCESS 1`] = `
Object {
  "error": null,
  "loading": false,
  "results": Array [
    "some",
    "results",
  ],
}
`;

exports[`OrganizationProducts reducer should return the initial state 1`] = `
Object {
  "error": null,
  "loading": false,
  "results": Array [],
}
`;
